packages/sign-client/test/sdk/client.spec.ts

```diff
@@ -306,6 +310,44 @@ describe("Sign Client Integration", () => {
         await deleteClients(clients);
       });
     });
+    describe("request queue", () => {
+      it("should reset request queue state on disconnect", async () => {
+        const {
+          clients,
+          sessionA: { topic },
+        } = await initTwoPairedClients({}, {}, { logger: "error" });
+        await new Promise<void>((resolve) => {
+          clients.B.once("session_request", () => {
+            resolve();
+          });
+          clients.A.request({
+            topic,
+            ...TEST_REQUEST_PARAMS,
+          });
+        });
+
+        expect(clients.B.pendingRequest.getAll().length).to.eq(1);
+        // @ts-expect-error - sessionRequestQueue is private property
```

Comment:
Why does this even successfully de-reference if it's a `private` property of `Engine`? 🤔